spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/RemoveOrphanFilesProcedure.java

```diff
@@ -120,6 +147,12 @@ public InternalRow[] call(InternalRow args) {
       if (fileListView != null) {
         action.compareToFileList(spark().table(fileListView));
       }
+      Preconditions.checkArgument(prefixMismatchMode == null ||
+              Lists.newArrayList("ignore", "error").contains(prefixMismatchMode.toLowerCase()),
```

Comment:
We should also support delete. What about adding `fromString` to `PrefixMismatchMode` and using it?

Reply:
```
      Preconditions.checkArgument(prefixMismatchMode == null ||
                      Arrays.stream(DeleteOrphanFiles.PrefixMismatchMode.values()).anyMatch(mode ->
                              mode.name().equals(prefixMismatchMode)),
```
Changed to above